src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ReflectionMethodBodyScanner.cs

```diff
@@ -23,7 +23,7 @@ public static bool ResolveType(string name, ModuleDesc callingModule, TypeSystem
             StringBuilder typeName = new StringBuilder();
             StringBuilder typeNamespace = new StringBuilder();
             string containingTypeName = null;
-            while (i < name.Length && (char.IsLetterOrDigit(name[i]) || name[i] == '.' || name[i] == '`' || name[i] == '+'))
+            while (i < name.Length && (char.IsLetterOrDigit(name[i]) || name[i] == '.' || name[i] == '_' || name[i] == '`' || name[i] == '+'))
```

Comment:
Not sure what the style conventions are in the compiler, but you could simplify this with a pattern if that's allowed?
```suggestion
            while (i < name.Length && (char.IsLetterOrDigit(name[i]) || name[i] is '.' or '_' or '`' or '+'))
```

Reply:
No need to polish this; this file will be nuked from orbit with #72833. This parser is kind of garbage. It's from a time when this being a heuristic was enough, as the comment above says. That time has passed.